File: SDKs/amazonaws.com/cloudfront/2018-11-05/python/src/sdk/models/shared/contenttypeprofile.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from . import *


@dataclass
class ContentTypeProfile:
    r"""ContentTypeProfile
    A field-level encryption content type profile. 
    """
    
    content_type: str = field()
    format: FormatEnum = field()
    profile_id: Optional[str] = field(default=None)
    

File: SDKs/amazonaws.com/cloudfront/2019-03-26/typescript/src/sdk/models/shared/contenttypeprofile.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { FormatEnum } from "./formatenum";



// ContentTypeProfile
/** 
 * A field-level encryption content type profile. 
**/
export class ContentTypeProfile extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  format: FormatEnum;

  @SpeakeasyMetadata()
  profileId?: string;
}
